### DISCUSSION TITLE
1인칭 카메라 구현

### DISCUSSION THREAD

**Discussion by Park-sechang** [naverz/zepeto-studio-kor#345] [5. Q&A]
1인칭 카메라를 구현 중인데, 제페토 카메라가 아닌 새로운 카메라를 만들어서 구현하려 합니다.

카메라가 제페토 캐릭터를 따라다니는 것은 되었지만,

1. 모바일 화면을 클릭하여 카메라 이동
2. 좌,우,뒤로 이동할때 걷기

이 두가지 기능을 어떻게 구현할지 잘 모르겠습니다. 

혹시 관련해서 아시는 분 있으실까요...???

**Comment by minjujuu** [comment]
1. 유니티에서 지원하는 Input.GetMouseButton() 류의 함수를 이용하여 Input.mousePosition 값을 얻어와서 
카메라 이동/회전에 적용해보는 건 어떨까요?
해당 함수 사용한 유니티 예제 참고해서 ZEPETO Script로 구현해봤는데 앱에서도 잘 적용되었습니다

2. 이동할 때 걷는 건 다음과 같이 해보시면 어떨까요?
[제페토 캐릭터 애니메이션 수정 관련](https://github.com/naverz/zepeto-script-sample/issues/65) -> 해당 링크 참고해서 해보니까
커스텀 할 Zepeto Animator 만든 다음 'Run' 에 적용된 모션을 'walk'로 변경하면
이동할 때도 걷는 애니메이션을 재생할 수 있었습니다
다만 옆으로 걷기, 뒤로 걷기는 여기서 더 처리해주어야 하거나, 아직 어려울 것 같네요..!
![image](https://user-images.githubusercontent.com/79794414/153164733-918de020-9213-4a02-a649-9dbce41a95b5.png)

**Reply by Park-sechang** [reply]
정말 많은 도움이 됐습니다.
답변 감사합니다!

**Reply by Park-sechang** [reply]
input.getmousebutton() 구현중에 이런 에러가 떴습니다.

![스크린샷(311)_LI](https://user-images.githubusercontent.com/99107908/153344955-1f8133eb-816d-4008-ab6b-4f357f4f5641.jpg)

c# 유니티에서 잘 작동하는데, 제페토 유니티에서는 작동하지 않네요.

혹시 왜 그러신지 아실까요..???

**Reply by minjujuu** [reply]
혹시 코드를 볼 수 있을까요?
정말 단순한 예제이긴 하지만 이런 식으로 사용했을 때 에러는 뜨지 않았어서요~

import { Input } from 'UnityEngine'
import { ZepetoScriptBehaviour } from 'ZEPETO.Script'

export default class Test extends ZepetoScriptBehaviour {

    Update()
    {
        if(Input.GetMouseButton(0)) {
            console.log("click");
        }
    }
}

**Reply by Park-sechang** [reply]
아 넵 코드입니다.

![스크린샷(312)](https://user-images.githubusercontent.com/99107908/153346927-67a68e09-2ea7-45f7-ab00-16fef25c2dee.png)

**Reply by Park-sechang** [reply]
보여주신대로 로그만 찍어봤는데 같은 에러가 뜹니다.

제가 다른 부분에서 놓치고 있는게 아닌가 싶네요..

**Reply by minjujuu** [reply]
음 혹시 Project Settings에서 Active Input Handling을 Both로 변경해주셨을까요?
![image](https://user-images.githubusercontent.com/79794414/153347717-bf34bb89-711d-42c7-b740-c2b6b6a6ae28.png)

**Reply by Park-sechang** [reply]
네. 시도해봤었지만 이런 에러가 떠서 다시 되돌렸습니다.

![스크린샷(313)](https://user-images.githubusercontent.com/99107908/153350228-a89e3ecf-2ffa-478e-9e6e-01a993b7097e.png)

하지만 다시 both로 하고 해보니 console.log는 작동하네요!!

아무래도 if (input.getmousebutton(0)) 안의 코드에서 에러가 나는거 같은데, 정말 죄송하지만 minjujuu님께서는 어떻게 구현하셨는지 알 수 있을까요..?

**Reply by minjujuu** [reply]
말씀주신 에러 재현해봤는데,

    private rx: float = 0;
    private ry: float = 0;

위처럼 초기화해주고 다시 실행하니까 해결되는 거 같아요
한 번 시도해보실래요?

**Reply by Park-sechang** [reply]
오 해결됐어요!! 생각보다 너무 간단한 문제였네요..ㅋㅋㅋㅋ

신경써주셔서 정말 감사합니다.
